File: LeetCode75/26_Remove_Duplicates_from_Sorted_Array/26_Remove_Duplicates_from_Sorted_Array.kt
class Solution {
    fun removeDuplicates(nums: IntArray): Int {
        if (nums.isEmpty()) return 0

        var k = 0

        for (i in nums.indices) {
            if (i == 0 || nums[i] != nums[k - 1]) {
                nums[k] = nums[i]
                k++
            }
        }
        return k
    }
}

/*
[アプローチ]
・重複している要素を1つにして、配列の要素数は入力値と同じにする
・配列の要素で、重複している要素を1つにする
 */

File: LeetCode75/26_Remove_Duplicates_from_Sorted_Array/26_Remove_Duplicates_from_Sorted_Array.ts
function removeDuplicates(nums: number[]):number {
    if (nums.length === 0) return 0

    let k = 0

    for (let i = 0; i < nums.length; i++) {
        if (i === 0 || nums[i] !== nums[k-1]) {
            nums[k] = nums[i]
            k++
        }
    }
    return k
}

const nums = [1,1,2,2]
console.log(removeDuplicates(nums))
